### PR TITLE
Add more detail about what hosts are allowed

### LINKED ISSUE #41545: config.hosts no longer supports underscores in host name and error message is incorrect
### Steps to reproduce
Create an application with an underscore in the hostname (e.g. rails_example). Add that hostname to config.hosts in your config/environment.rb. Update your local hosts file to point your hostname to 127.0.0.1 so that your local dns will resolve your hostname to your locally running server. When navigating to https://rails_example:3000, you will see the following error:

```
Blocked host: rails_example
To allow requests to rails_example, add the following to your environment configuration:
config.hosts << "rails_example"
```

```ruby
# config/environment.rb
Rails.application.configure do
...
# returns error saying to add "rails_example" to config.hosts
config.hosts << "rails_example"
...
end
```
This issue is due to an [updated regex](https://github.com/rails/rails/compare/v6.0.3.4...v6.0.3.5#diff-401b1f2d4b94c52328880f3baca952f374f24903245327fc4c4527d6d5655a0cR90-R95) for hostname validation introduced in Rails 6.0.3.5 which no longer allows underscores in the hostname. 

### Expected behavior
User should receive a warning that the hostname they specified in config.hosts is not a valid hostname.

### Actual behavior
User is told to add the same host name that they have already added to config.hosts.

### System configuration
**Rails version**: 6.0.3.5

**Ruby version**: 2.6.6

### PR DESCRIPTION
**Pull request by ghiculescu**
Fixes https://github.com/rails/rails/issues/41545

### CODE REVIEW

File: actionpack/lib/action_dispatch/middleware/templates/rescues/blocked_host.text.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Blocked host: <%= @host %>
 
-To allow requests to <%= @host %>, add the following to your environment configuration:
+To allow requests to <%= @host %> make sure it is a valid hostname (containing only numbers, letters, dashes and dots), then add the following to your environment configuration:
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/middleware/templates/rescues/blocked_host.text.erb:3]: We also need to update the HTML version of this.

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: actionpack/lib/action_dispatch/middleware/templates/rescues/blocked_host.text.erb (hunk 1 above)
Comment (ghiculescu) [reply actionpack/lib/action_dispatch/middleware/templates/rescues/blocked_host.text.erb:3]: Oops, yep.

**Comment by ghiculescu** [comment]
What do you think about displaying the actual regex in the error page? Something like this:

```html
<small>The hostname needs to match this regex: <pre><%= ActionDispatch::HostAuthorization::HOSTNAME.inspect =></pre></small>
```

**Comment by rafaelfranca** [comment]
I think the plain text version of the regexp is better, that decrease the cognitive load to understand what needs to be done.

**Pull request merged**